compiler-rt/lib/mpk_untrusted/tests/unit/alloc_site_handler_tests.cpp

```diff
@@ -0,0 +1,85 @@
+// relocate in lib
+#include "../util/mpk_untrusted_test_config.h"
+#include "alloc_site_handler.h"
+#include "gtest/gtest.h"
+#include <limits>
+#include <memory>
+
+namespace __mpk_untrusted {
+
+// For this test to work properly, we have to ensure all other tests
+// remove all allocations that they have added to the map.
+TEST(getAllocSite, EmptyMap) {
+  auto handle = AllocSiteHandler::get();
+  auto allocSite = handle->getAllocSite(nullptr);
+  EXPECT_EQ(allocSite.get(), AllocSite::error().get());
+}
+
+TEST(getAllocSite, InvalidPreAddress) {
+  auto handle = AllocSiteHandler::get();
+  auto ptr = (rust_ptr)malloc(sizeof(uint64_t));
+  handle->insertAllocSite(
+      ptr, std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1));
+  auto decPtr = (rust_ptr)((uintptr_t)ptr - 1);
+  auto allocSite = handle->getAllocSite(decPtr);
+  EXPECT_EQ(allocSite.get(), AllocSite::error().get());
+  handle->removeAllocSite(ptr);
+}
+
+TEST(getAllocSite, InvalidPostAddress) {
+  auto handle = AllocSiteHandler::get();
+  auto ptr = (rust_ptr)malloc(sizeof(uint64_t));
+  handle->insertAllocSite(
+      ptr, std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1));
+  auto incPtr = (rust_ptr)((uintptr_t)ptr + sizeof(uint64_t));
+  auto allocSite = handle->getAllocSite(incPtr);
+  EXPECT_EQ(allocSite.get(), AllocSite::error().get());
+  handle->removeAllocSite(ptr);
+}
+
+TEST(getAllocSite, InvalidBetweenAddress) {
+  auto handle = AllocSiteHandler::get();
+  auto ptr = (rust_ptr)malloc(sizeof(uint64_t));
+  auto ptr2 = (rust_ptr)((uintptr_t)ptr + (sizeof(uint64_t) * 2));
+  handle->insertAllocSite(
+      ptr, std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1));
+  handle->insertAllocSite(
+      ptr2, std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1));
+  auto decPtr = (rust_ptr)((uintptr_t)ptr + (sizeof(uint64_t) * 1));
+  auto allocSite = handle->getAllocSite(decPtr);
+  EXPECT_EQ(allocSite.get(), AllocSite::error().get());
+  handle->removeAllocSite(ptr);
+  handle->removeAllocSite(ptr2);
+}
+
+TEST(getAllocSite, ValidBetweenAddress) {
+  auto handle = AllocSiteHandler::get();
+  auto ptr = (rust_ptr)malloc(sizeof(uint64_t));
+  auto ptr2 = (rust_ptr)((uintptr_t)ptr + (sizeof(uint64_t) * 4));
+  auto newAllocSite = std::make_shared<AllocSite>(ptr, sizeof(uint64_t) * 2, 1);
+  handle->insertAllocSite(ptr, newAllocSite);
+  handle->insertAllocSite(
+      ptr2, std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1));
+  auto decPtr = (rust_ptr)((uintptr_t)ptr + (sizeof(uint64_t) * 1));
+  auto getAllocSite = handle->getAllocSite(decPtr);
+  EXPECT_EQ(getAllocSite.get(), newAllocSite.get());
+  handle->removeAllocSite(ptr);
+  handle->removeAllocSite(ptr2);
+}
+
+TEST(getAllocSite, Overflow) {
+  auto ptr = (rust_ptr)std::numeric_limits<unsigned int>::max();
+  auto newAllocSite = std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1);
+  EXPECT_TRUE(newAllocSite->containsPtr(++ptr));
```

Comment:
can you explain this test?
 you have a pointer at 0xffffff... w/ size 4. so the end of the addr range is 0x3?;
internally, the containsPtr is doing arithmetic, but then you also do pointer arithmetic below.
the incremented pointer should point at 0x3 if it wraps. why is containsPtr returning true? this doesn't seem correct.

Reply:
even if it is correct, the test should be a bit more explicit.
for example it should probably have a name like PointerArithmeticOverflowWraps.
Also, its should be a bit easier to see what pointer values are being tested.
something like
```C++
void* ptr = (void*)(-1); // 0xffffffff...
```
the exact way its done isn't important, but that its easy to understand from the assertion that the value was expected to wrap.